cryptolib/verifier.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "errors"
+
+// Verifier contains methods to validate an Attestation.
+type Verifier interface {
+	VerifyAttestation(att *Attestation, publicKey []byte, image string) error
```

Comment:
I would prefer to see this split into two layers: 

1. Low level cryptographic authentication (i.e. `att`'s signature is validated by `publicKey`).
2. High level verification in the context of the current policy evaluation (i.e. `att` is both cryptographically authenticated *and* its payload correctly verifies `image` according to Kritis semantics).

(2) would call (1) as an implementation detail.

Reply:
Done. We added `verifyPkix`, `verifyPgp`, and `verifyJwt` functions to explicitly highlight the low level crypto layer.